src/core/probe/manager.rs

```diff
@@ -249,9 +229,13 @@ impl ProbeManager {
 
         // First, handle generic probes.
         let mut set = ProbeSet {
-            probes: self.generic_probes.clone(),
+            probes: self
+                .probes
+                .clone()
+                .into_iter()
+                .filter(|(_, p)| p.hooks_len() == 0)
```

Comment:
Maybe add a helper function in `Probe` such as `is_generic() -> bool` so that these filters are easier to understand?